packages/fxa-settings/src/components/RecoveryKeySetupHint/index.tsx

```diff
@@ -0,0 +1,181 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useEffect, useState } from 'react';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { Control, useForm, useWatch } from 'react-hook-form';
+import InputText from '../InputText';
+import { useFtlMsgResolver } from '../../models';
+import { logViewEvent } from '../../lib/metrics';
+import { LightbulbImage } from '../images';
+import { DISPLAY_SAFE_UNICODE } from '../../constants';
+import Banner, { BannerType } from '../Banner';
+import {
+  AuthUiErrorNos,
+  AuthUiErrors,
+} from '../../lib/auth-errors/auth-errors';
+import classNames from 'classnames';
+import { getErrorFtlId } from '../../lib/error-utils';
+
+export type RecoveryKeySetupHintProps = {
+  updateRecoveryKeyHint: (hint: string) => Promise<void>;
+  navigateForward: () => void;
+  viewName: string;
+};
+
+type FormData = { hint: string };
+
+export const MAX_HINT_LENGTH = 255;
+
+export const RecoveryKeySetupHint = ({
```

Comment:
A large chunk of this was moved from `FlowRecoveryKeyHint` and tweaked to work with both page components. I left the flow metrics/`viewName` stuff as-is.

I did modify the top/bottom padding of `LightbulbImage` though, it was too much space based on both Figmas.